File: CS/HowToRemoveNewButton.Module/MyController.cs
using System;
using System.ComponentModel;
using System.Collections.Generic;
using System.Diagnostics;
using System.Text;

using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Actions;
using DevExpress.Persistent.Base;

using DevExpress.ExpressApp.SystemModule;

namespace HowToRemoveNewButton.Module {
   public partial class MyController : ViewController {
      public MyController() {
         InitializeComponent();
         RegisterActions(components);
      }

      private void MyController_FrameAssigned(object sender, EventArgs e) {
         Frame.GetController<NewObjectViewController>().Active.SetItemValue("Custom", IsNewActionEnabled());
      }
      //Define conditions under which the NewObjectViewController must be activated/deactivated
      private bool IsNewActionEnabled() {
         return (View is ListView) && View.IsRoot;
      }

      private void MyController_Activated(object sender, EventArgs e) {
         //By default, the List Views that are displayed in Lookup Editors contain "LookupListView" in their names
         if (View.Id.EndsWith("LookupListView"))
         {
            NewObjectViewController standardController = Frame.GetController<NewObjectViewController>();
            if (standardController != null)
               standardController.Active.SetItemValue("LookupListView", false);
         }
      }
   }
}

File: VB/HowToRemoveNewButton.Module/MyController.vb
Imports System
Imports System.ComponentModel
Imports System.Collections.Generic
Imports System.Diagnostics
Imports System.Text

Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Actions
Imports DevExpress.Persistent.Base

Imports DevExpress.ExpressApp.SystemModule

Public Class MyController
	Inherits DevExpress.ExpressApp.ViewController

	Public Sub New()
		MyBase.New()

		'This call is required by the Component Designer.
		InitializeComponent()
		RegisterActions(components) 

	End Sub

    Private Sub MyController_Activated(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Activated
        'By default, the List Views displayed in Lookup Editors contain "LookupListView" in their names
        If View.Id.EndsWith("LookupListView") Then
            Dim standardController As NewObjectViewController = Frame.GetController(Of NewObjectViewController)()
            If Not standardController Is Nothing Then
                standardController.Active.SetItemValue("LookupListView", False)
            End If
        End If
    End Sub

    Private Sub MyController_FrameAssigned(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.FrameAssigned
        Frame.GetController(Of NewObjectViewController)().Active.SetItemValue("Custom", IsNewActionEnabled())
    End Sub
    'Define conditions under which the NewObjectViewController must be activated/deactivated
    Private Function IsNewActionEnabled() As Boolean
        Return (TypeOf View Is ListView) AndAlso View.IsRoot
    End Function
End Class
